From 3eb322518ff8d3fab29cc39f8180a88ffdc79ecf Mon Sep 17 00:00:00 2001
From: Mykyta Biliavskyi <m.biliavskyi@samsung.com>
Date: Tue, 5 Jul 2016 20:18:32 +0300
Subject: [PATCH 76/79] Edje_Edit: allow set NULL as a style name for textblock
 part.

NULL as 'style' value remove style name from textblock part state.
---
 src/lib/edje/Edje_Edit.h |  3 ++-
 src/lib/edje/edje_edit.c | 10 +++++++++-
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/lib/edje/Edje_Edit.h b/src/lib/edje/Edje_Edit.h
index 696af89..9fb682f 100644
--- a/src/lib/edje/Edje_Edit.h
+++ b/src/lib/edje/Edje_Edit.h
@@ -6112,7 +6112,8 @@ EAPI const char * edje_edit_state_text_style_get(Evas_Object *obj, const char *p
  * @param part Part to set the style of.
  * @param state State in which the style is set.
  * @param value Value of the state.
- * @param style The style name to use.
+ * @param style The style name to use. In case when NULL style will removed from textblock part description.
+ *
  *
  * @return @c EINA_TRUE in case of success, @c EINA_FALSE otherwise.
  */
diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index 8282700..c1a2fef 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -8463,7 +8463,15 @@ edje_edit_state_text_style_set(Evas_Object *obj, const char *part, const char *s
      return EINA_FALSE;
 
    txt = (Edje_Part_Description_Text *)pd;
-   _edje_if_string_replace(ed, &txt->text.style.str, style);
+   if (style == NULL )
+     {
+        _edje_if_string_free(ed, &txt->text.style.str);
+        txt->text.style.str = NULL;
+     }
+   else
+     {
+        _edje_if_string_replace(ed, &txt->text.style.str, style);
+     }
 
    edje_object_calc_force(obj);
    return EINA_TRUE;
-- 
2.7.4

